components/scream/src/share/atm_process/atmosphere_process_group.cpp

```diff
@@ -184,6 +184,12 @@ gather_internal_fields  () {
 void AtmosphereProcessGroup::initialize_impl (const RunType run_type) {
   for (auto& atm_proc : m_atm_processes) {
     atm_proc->initialize(timestamp(),run_type);
+#ifdef SCREAM_HAS_MEMORY_USAGE
+    long long my_mem_usage = get_mem_usage(MB);
+    long long max_mem_usage;
+    m_comm.all_reduce(&my_mem_usage,&max_mem_usage,1,MPI_MAX);
+    m_atm_logger->info("[EAMxx::initialize::"+atm_proc->name()+"] memory usage: " + std::to_string(max_mem_usage) + "MB");
```

Comment:
Here too. The AD already reports a total ATM memory usage right after all atm procs are inited. This is more fine grained, which probably is better suited for "debug" level.